### ISSUE TITLE
Tekton Pact stage in igc-nodejs-v1-23-0 fails: Cannot find module '../util/trace-constants'

### ISSUE THREAD

**Issue by bwoolf1** [ibm-garage-cloud/planning#424]
**Describe the bug**
Deploying a microservice based on [template-node-typescript](https://github.com/IBM/template-node-typescript) using Tekton pipeline igc-nodejs-v1-23-0. The first stage, which is the test stage, fails; so the whole pipeline fails. The code is unchanged from the original template except for renaming the app in package.json.

**To Reproduce**
1. Follow the steps in [Deploying an App](https://cloudnativetoolkit.dev/getting-started/deploy-app/).
2. Choose the Typescript Microservice template.
3. Choose the igc-nodejs-v1-23-0 Tekton pipeline
4. The OpenShift console shows the pipeline run fails on the first stage:
```
> inventory-svc-stock@1.0.0 build /workspace/source
> tsc
src/logger/logger-pino.service.ts(6,30): error TS2307: Cannot find module '../util/trace-constants'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! inventory-svc-stock@1.0.0 build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the inventory-svc-stock@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Expected behavior**
The pipeline should complete successfully and deploy the app.

**Screenshots**
Here's the complete log for the test stage, which is the only stage that ran:
```
STEP-GIT-SOURCE-INVENTORY-SVC-STOCK-GIT-LGN2S

{"level":"info","ts":1593121153.0128112,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1593121157.045876,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
{"level":"info","ts":1593121157.6244726,"logger":"fallback-logger","caller":"git/git.go:102","msg":"Successfully cloned https://github.com/devadv-sandbox-team-one/inventory-svc-stock.git @ master in path /workspace/source"}
{"level":"info","ts":1593121157.820396,"logger":"fallback-logger","caller":"git/git.go:130","msg":"Successfully initialized and updated submodules in path /workspace/source"}
STEP-BUILD

{"level":"info","ts":1593121153.4802394,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}

> dtrace-provider@0.8.8 install /workspace/source/node_modules/dtrace-provider
> node-gyp rebuild || node suppress-error.js

make: Entering directory '/workspace/source/node_modules/dtrace-provider/build'
  TOUCH Release/obj.target/DTraceProviderStub.stamp
make: Leaving directory '/workspace/source/node_modules/dtrace-provider/build'

> core-js@2.6.11 postinstall /workspace/source/node_modules/babel-runtime/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"

[96mThank you for using core-js ([94m https://github.com/zloirock/core-js [96m) for polyfilling JavaScript standard library![0m

[96mThe project needs your help! Please consider supporting of core-js on Open Collective or Patreon: [0m
[96m>[94m https://opencollective.com/core-js [0m
[96m>[94m https://www.patreon.com/zloirock [0m

[96mAlso, the author of core-js ([94m https://github.com/zloirock [96m) is looking for a good job -)[0m


> core-js@3.6.5 postinstall /workspace/source/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> @pact-foundation/pact-node@10.9.0 postinstall /workspace/source/node_modules/@pact-foundation/pact-node
> node postinstall.js

Installing Pact Standalone Binary for linux.
Downloading Pact Standalone Binary v1.82.3 for platform linux from https://github.com/pact-foundation/pact-ruby-standalone/releases/download/v1.82.3/pact-1.82.3-linux-x86_64.tar.gz
Please note: we are tracking this download anonymously to gather important usage statistics. To disable tracking, set 'pact_do_not_track: true' in your package.json 'config' section.
Finished downloading binary to /workspace/source/node_modules/@pact-foundation/pact-node/standalone/pact-1.82.3-linux-x86_64.tar.gz
Extracting binary from /workspace/source/node_modules/@pact-foundation/pact-node/standalone/pact-1.82.3-linux-x86_64.tar.gz.
Checksum passed for 'pact-1.82.3-linux-x86_64.tar.gz'.
Extraction done.


### If you ❤ Pact and want to support us, please donate here: http://donate.pact.io/node


Pact Standalone Binary is ready.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})

added 1446 packages from 778 contributors and audited 1457 packages in 38.997s

23 packages are looking for funding
  run `npm fund` for details

found 1 low severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details

> inventory-svc-stock@1.0.0 prebuild /workspace/source
> npm run clean


> inventory-svc-stock@1.0.0 clean /workspace/source
> rimraf dist


> inventory-svc-stock@1.0.0 build /workspace/source
> tsc

src/logger/logger-pino.service.ts(6,30): error TS2307: Cannot find module '../util/trace-constants'.
npm ERR! code ELIFECYCLE
npm ERR! errno 2
npm ERR! inventory-svc-stock@1.0.0 build: `tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the inventory-svc-stock@1.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /tekton/home/.npm/_logs/2020-06-25T21_40_09_114Z-debug.log
STEP-TEST

{"level":"info","ts":1593121154.0055976,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
2020/06/25 21:40:10 Skipping step because a previous step failed
STEP-PUBLISH-PACT

{"level":"info","ts":1593121154.4790392,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
2020/06/25 21:40:10 Skipping step because a previous step failed
STEP-VERIFY-PACT

{"level":"info","ts":1593121154.9355133,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
2020/06/25 21:40:10 Skipping step because a previous step failed
STEP-SONAR-SCAN

{"level":"info","ts":1593121155.332577,"logger":"fallback-logger","caller":"logging/config.go:69","msg":"Fetch GitHub commit ID from kodata failed: \"KO_DATA_PATH\" does not exist or is empty"}
2020/06/25 21:40:11 Skipping step because a previous step failed
```

**IBM Cloud**
Select the services and tools affected

- [ ] IBM Kubernetes Service Managed Service
- [ ] Red Hat OpenShift Managed Services
- [ ] Jenkins
- [X] Tekton
- [ ] SonarQube
- [X] Pact
- [ ] Artifactory
- [ ] IBM Image Registry

**Comment by bwoolf1** [comment]
Meanwhile, an app based on [Spring Boot Microservice](https://github.com/IBM/template-java-spring) deployed with pipeline igc-java-gradle-v1-23-0 has deployed properly.

**Comment by mjperrins** [comment]
Can you compare the code base from the base starter kit and the inventory version for changes and maybe engage @lsteck to have a look , I have a set of inventory apps from a month ago that are deploying clean, trying and work out the change history thanks

**Comment by bwoolf1** [comment]
@seansund committed https://github.com/IBM/template-node-typescript/commit/4f30cb952fc1ccb6cc55d0f109d3fc158deeaeb7 two days ago, which moved the file trace-constants from src/util/ to src/util/opentracing/. Two references in the change set look correct to me, but there must be another reference that's still looking for the file in the old path.

**Comment by bwoolf1** [comment]
https://github.com/IBM/template-node-typescript/issues/7 fixes two problems in `src/logger/logger-pino.service.ts`. Now `STEP-BUILD` works.

Now `STEP-TEST` fails:
```
> typescript-fix@1.0.0 test /workspace/source
> jest --coverage

jest-haste-map: Haste module naming collision: typescript-fix
  The following files share their name; please adjust your hasteImpl:
    * <rootDir>/package.json
    * <rootDir>/dist/package.json

FAIL test/util/opentracing/express-middleware.spec.ts (5.638s)
  ● express-middelware › given buildTraceContext() › when context contains uber-trace-id › then return {traceId, spanId}

    expect(received).toEqual(expected) // deep equality

    - Expected  - 0
    + Received  + 2

      Object {
    +   "flags": "0",
    +   "parentSpanId": "0",
        "spanId": "spanid",
        "traceId": "traceid",
      }

      28 |         const context = {'uber-trace-id': `${traceId}:${spanId}:0:0`};
      29 | 
    > 30 |         expect(buildTraceContext(context)).toEqual({traceId, spanId});
         |                                            ^
      31 |       });
      32 |     });
      33 |   })

      at test/util/opentracing/express-middleware.spec.ts:30:44
      at node_modules/tslib/tslib.js:113:75
      at Object.__awaiter (node_modules/tslib/tslib.js:109:16)
      at Object.<anonymous> (test/util/opentracing/express-middleware.spec.ts:24:56)
```

As shown, the file `test/util/opentracing/express-middleware.spec.ts`, line 30, indeed is:
```
expect(buildTraceContext(context)).toEqual({traceId, spanId});
```

So that equality test doesn't run successfully. I have no idea how to fix this test.

**Comment by bwoolf1** [comment]
I think I've fixed the test. The pipeline completes successfully now, so I think everything is working.
